Integrations/integration-CrowdStrikeFalcon.yml

```diff
@@ -0,0 +1,1095 @@
+                    if arg[0]:
+                        payload[k] = arg
+                elif arg:
+                    payload[k] = arg
+            ids = http_request('GET', '/indicators/queries/iocs/v1', payload).get('resources')
+            if not ids:
+                return None
+        else:
+            ids = str(ids)
+        payload = {
+            'ids': ids
+        }
+        return http_request('GET', '/indicators/entities/iocs/v1', params=payload)
+
+
+    def enrich_ioc_dict_with_ids(ioc_dict):
+        """
+            Enriches the provided ioc_dict with IoC ID
+            :param ioc_dict: IoC dict transformed using the SEARCH_IOC_KEY_MAP
+            :return: ioc_dict with its ID key:value updated
+        """
+        for ioc in ioc_dict:
+            ioc['ID'] = '{0}:{1}'.format(ioc.get('Type'), ioc.get('Value'))
+        return ioc_dict
+
+
+    def delete_ioc():
+        """
+            UNTESTED - Sends a delete IoC request
+            :return: Response json of delete IoC
+        """
+        ids = str(demisto.args().get('ids'))
+        payload = {
+            'ids': ids
+        }
+        return http_request('DELETE', '/indicators/entities/iocs/v1', payload)
+
+
+    def update_iocs():
+        """
+            UNTESTED - Updates the values one or more IoC
+            :return: Response json of update IoC request
+        """
+        args = demisto.args()
+        input_args = {
+            'ids': args.get('ids'),
+            'policy': args.get('policy', ''),
+            'expiration_days': args.get('expiration_days', ''),
+            'source': args.get('source'),
+            'description': args.get('description')
+        }
+        payload = {k: str(v) for k, v in input_args.items() if v}
+        headers = {'Authorization': HEADERS['Authorization']}
+        return http_request('PATCH', '/indicators/entities/iocs/v1', params=payload, headers=headers)
+
+
+    def search_device():
+        """
+            Searches for devices using the argument provided by the command execution. Returns empty
+            result of no device was found
+            :return: Search device response json
+        """
+        args = demisto.args()
+        input_arg_dict = {
+            'device_id': str(args.get('ids', '')).split(','),
+            'status': str(args.get('status', '')).split(','),
+            'hostname': str(args.get('hostname', '')).split(','),
+            'platform_name': str(args.get('platform_name', '')).split(','),
+            'site_name': str(args.get('site_name', '')).split(',')
+        }
+        url_filter = ''.format(str(args.get('filter', '')))
+        for k, arg in input_arg_dict.items():
+            if arg:
+                if type(arg) is list:
+                    arg_filter = ''
+                    for arg_elem in arg:
+                        if arg_elem:
+                            arg_filter = "{0}:'{1}'".format('{0},{1}'.format(arg_filter, k) if arg_filter else k, arg_elem)
```

Comment:
please use named place holders when formatting more than one variable in a string.